api/v1alpha1/hostedcluster_types.go

```diff
@@ -155,12 +156,14 @@ type HostedClusterSpec struct {
 	//
 	// +optional
```

Comment:
I think we also need a type scoped rule here to validate once set it's not removed, e.g. https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutablility-after-first-modification

Reply:
So everything that is marked as immutable and optional should follow this pattern then right?

Reply:
yeh correct, which is the case for everything we default. That's because of a limitation on the api machinery open api and clients which also run validation but not defaulting so if you try to kubectl apply a required field with a default without passing a explicit value, it would fail validation.

Feel free to skip the type level validation to prevent setting back to empty. We can go in steps.